### PR TITLE
Layout manager docs and debug code

### PR DESCRIPTION
**Pull request by asturur**
## Description

This PR provides documentation for the layour manager pr since there isn't much.
Currently in the process of understanding that code so i need it documented

## Changes

- some variable names have been changed

### CODE REVIEW

**Comment by asturur** [comment]
this is first of a series, i ll commit docs everytime i can sit and write some.

File: src/LayoutManager/LayoutStrategies/utils.ts [hunk 1]
```diff
@@ -1,56 +1,95 @@
...
-    const rad = degreesToRadians(object.angle),
-      sine = Math.abs(sin(rad)),
-      cosine = Math.abs(cos(rad)),
-      rx = sizeVector.x * cosine + sizeVector.y * sine,
-      ry = sizeVector.x * sine + sizeVector.y * cosine;
-    return new Point(rx, ry);
+  if (decomposition.angle !== 0) {
+    return compensateAngle(object.angle, sizeVector);
   }
   return sizeVector;
```
Comment (ShaMan123) [inline src/LayoutManager/LayoutStrategies/utils.ts:45]: same - inline

File: src/LayoutManager/LayoutStrategies/utils.ts [hunk 2]
```diff
@@ -1,56 +1,95 @@
...
+//     ctx.stroke();
+//   }, 50);
+// };
+
 /**
- * account for selected objects in active selection
+ * Return 2 points, the topLeft corner and the bottom right corner of an object
+ * that needs to go inside group. This needs to take in account objects that are in
+ * a group but also in an active selection or when they are about to be moved from a
+ * group to another
```
Comment (ShaMan123) [inline src/LayoutManager/LayoutStrategies/utils.ts:73]:
```suggestion
* a group but also in an active selection.
```

File: src/LayoutManager/LayoutStrategies/utils.ts [hunk 3]
```diff
@@ -1,56 +1,95 @@
...
-    : object.getRelativeCenterPoint();
-  const sizeVector = getObjectSizeVector(object, planeChange);
+export const getObjectBounds = (
+  group: Group,
+  object: FabricObject
+): [Point, Point] => {
+  let objCenter = object.getRelativeCenterPoint();
+  let sizeVector;
+  // TODO: detail use cases
+  if (object.group && object.group !== group) {
```
Comment (ShaMan123) [inline src/LayoutManager/LayoutStrategies/utils.ts:82]: only if object is selected

File: src/LayoutManager/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Layout manager
+
+Layout manager exposes a main public method that perform group layouting given a target group and target objects.
```
Comment (ShaMan123) [inline src/LayoutManager/README.md:3]:
```suggestion
Layout manager exposes a main public method that performs group layouting 
```

File: src/LayoutManager/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+this is the call flow:
+
+performLayout -> onBeforeLayout -> getLayoutResult -> commitLayout -> onAfterLayout
+
+The flow has the following goals/features:
+
+- alerting the user a layout ends and finish
+- subscribe to object events in order to continue layouting later
+- calculate group new size and position
+- move the objects to their new position relative to the group center.
```
Comment (ShaMan123) [inline src/LayoutManager/README.md:24]:
```suggestion
- subscribe to objects' events in order to perform layout
- calculate group new size and position
- adjust objects' position relative to the new group center
- alerting the user a layout ends and finish
```

File: src/LayoutManager/README.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+**Note**: the option `objectsRelativeToGroup` is related to restoring a group from serialization and should be removed. Object should be eventually normalized in the from Object function or we should ask why do we want to perform a layout if the group has been saved with the layout properties already calculated.
```
Comment (ShaMan123) [inline src/LayoutManager/README.md:26]: a very good points
sounds we should not

File: src/LayoutManager/README.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+### type INITIALIZATION
+
+When a group is created, all objects gets flagged with the group property set to the new group, then an instance of a layout manager is used to calculate the bounding box.
```
Comment (ShaMan123) [inline src/LayoutManager/README.md:30]:
```suggestion
When a group is created, all objects get flagged with the group property
```

File: src/LayoutManager/README.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+commitLayout will get the above results and
+
+- set width/height of the group
+- set the position of all objects ( unless we are coming from `objectsRelativeToGroup`)
+- set the group top/left by either the passed in x/y or the nextCenter value
+
+onAfterLayout will
+
+- fire an user event
+- reset the group transform if the group is empty ( why? )
```
Comment (ShaMan123) [inline src/LayoutManager/README.md:54]: We need to decide on this
Consider having a group, adding objects, transforming
Then reomving the objects
Then adding an object.
The added object will be transformed - not sure about expectations.
For active selection epections are for the transform to reset.
#9087 => `e2e/tests/selection/stale-state/index.spec.ts`

**Comment by asturur** [comment]
@ShaMan123 this PR was just a way to insert some docs with UI commit squash, and i changed that code because i need to divide it to better /log udnerstand it.
I have more docs to write and questions to make, then we can put back the code to what it was or change it. The code changes i made aren't meant to be a review of your code or an improvement.

**Pull request merged**